### ISSUE TITLE
Heatmap plot in sobol analysis with groups

### ISSUE THREAD

**Issue by nikithiel** [SALib/SALib#612]
Hey there,

First of all, thanks for the great package! It integrates well into my coding project and helps me a lot.

R.n. I'm doing a Sobol analysis using grouping of my 24 input parameters. When I call `problem.heatmap()` after the analysis, I get the following heatmap, which I guess isn't exactly the right behavior:

![image](https://github.com/SALib/SALib/assets/42320487/704b6487-4e1d-4ca2-937e-de285870b9fb)

The heatmap shows the results for groups 2, 1, 3 and 4, but the labels are incorrect. In `heatmap.py`, `heatmap()` uses the following as x-axis labels:

https://github.com/SALib/SALib/blob/ee5e2cfc4193cdfc1edbfd3e8721f7140529d78e/src/SALib/plotting/heatmap.py#L71-L77

I think the correct behavior would be something like this:

```rb
groups = _check_groups(problem)

if not groups:
    namesGSA = problem['names']
else:
    namesGSA, _ = extract_group_names(groups)
```

Let me know what you think about it. Maybe I got something wrong here.

**Comment by ConnectedSystems** [comment]
Hi @nikithiel thanks for reporting this.

Is it possible to share a minimum working example? Or at least your problem specification? It would help confirm, or track down the cause.

Thank you!

**Comment by nikithiel** [comment]
Hey, thanks for checking the MWE. This indexing problem is exactly what I've meant. Results generally look the same for me.

**Comment by ConnectedSystems** [comment]
> Hey, thanks for checking the MWE. This indexing problem is exactly what I've meant. Results generally look the same for me.

Oh okay, I misunderstood. Thought the G1/G2 were the equivalent to x1, x2 etc, and that you were seeing groups being ignored.

The fix will be merged in shortly if you would like to try the dev version. I will release a patched version over the weekend.

**Comment by ConnectedSystems** [comment]
Addressed by #613 , thanks @nikithiel for reporting!
